src/components/Header/HeaderClient.tsx

```diff
@@ -72,20 +72,26 @@ export default function HeaderClient({
         <header className={`${className} fixed z-[9999] w-full`}>
             <ScrollShader />
             <div className="mx-auto mt-8 w-responsive">
-                <div className="flex items-center gap-8 border-4 border-black bg-white px-8 py-4 text-grey">
-                    <LogoTitle titleColor="text-grey" className="grow" />
-                    <Links />
-                    {data.nextStep === 'signup' && (
-                        <Button colour="purple" href="/join">
-                            Continue Signing Up
-                        </Button>
-                    )}
-                    {data.nextStep === 'payment' && (
-                        <Button colour="orange" href="/settings">
-                            Continue to payment
-                        </Button>
-                    )}
-                    {data.isSignedIn ? <UserButton data={data} /> : <SignInJoin />}
+                <div className="flex items-center justify-between gap-8 border-4 border-black bg-white px-8 py-4 text-grey">
+                    <div className="flex flex-shrink items-center">
+                        <LogoTitle titleColor="text-grey" className="grow" />
+                    </div>
+                    <div className="flex items-center md:gap-4">
+                        <Links />
+                        <div className="flex gap-4 md:flex-col md:justify-center lg:flex-row lg:items-center">
+                            {data.nextStep === 'signup' && (
+                                <Button colour="purple" href="/join">
+                                    Continue Signing Up
+                                </Button>
+                            )}
+                            {data.nextStep === 'payment' && (
+                                <Button colour="orange" href="/settings">
+                                    Continue to payment
+                                </Button>
+                            )}
+                            {data.isSignedIn ? <UserButton data={data} /> : <SignInJoin />}
+                        </div>
+                    </div>
                 </div>
                 <div className="relative -right-[0.5rem] -top-[4.75rem] -z-10 h-[5.25rem] w-responsive border-4 border-black bg-white" />
```

Comment:
Make the bottom rectangle in the header taller when the buttons go on top of each other 
<img width="620" alt="image" src="https://github.com/compsci-adl/website/assets/47909638/83dcc4f7-ad71-4cf7-997b-c33ee9c6bb30">

So that the spacing between the rectangles is consistent like this.
<img width="1150" alt="image" src="https://github.com/compsci-adl/website/assets/47909638/aeb47aec-90fd-4be6-b3a1-c394f7c60d60">